fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -5080,13 +5082,48 @@ Reference<TCServerInfo> DDTeamCollection::findOneLeastUsedServer() const {
 	}
 }
 
+bool DDTeamCollection::isAvailableToBuildMoreServerTeam(const TCMachineTeamInfo& machineTeam) const {
+	// This checking takes effects only if CHECK_SERVER_TEAM_WHEN_SELECT_MACHINE_TO_BUILD_SERVER_TEAM is set
+	ASSERT(SERVER_KNOBS->CHECK_SERVER_TEAM_WHEN_SELECT_MACHINE_TO_BUILD_SERVER_TEAM);
+	const int targetTeamNumPerServer =
```

Comment:
Nit: Can this formula be written as a standalone function and shared by the different callers?